wagtail_transfer/vendor/wagtail_admin_api/views.py

```diff
@@ -61,15 +61,6 @@ def get_base_queryset(self):
         """
         return Page.objects.all()
 
-    def get_queryset(self):
```

Comment:
Ping @gasman do you remember what this was used for? Is this obsolete?

Reply:
@kaedroho @gasman This is the last item in the reviews. I didnt find any specific purpose for this code (on the surface). Everything _seemed_ to work the same without it. Removing this removed an error that was being raised.

Reply:
Good question! I'm assuming that the bit of functionality at issue here is the exclusion of the root node, and from some deep digging it seems that the decision to do that was part of the admin API from day one (along with the todo note and [a promise to make it configurable in a future PR](https://github.com/wagtail/wagtail/pull/2507#discussion-diff-69732509) which evidently never happened :-) )

The call to `specific()` _is_ needed, so that custom `get_admin_display_title` methods on page subclasses are respected.

Since this is an internal vendored copy of the API we could go ahead and make any breaking changes we like here, but obviously it would be nice to keep it in sync with #5679 so that we can phase this copy out when that gets merged (and we drop support for older Wagtail versions). So the question becomes, is there anything in Wagtail that will break if we drop the `exclude(depth=1)`? I don't think there is - if I'm not mistaken, the only thing using the admin API is the explorer menu, and that only ever makes `child_of=` requests, so it can't ever return the root anyhow. But in that case, I'm a bit puzzled why we would have gone to the trouble of overriding the `wagtail.api.v2` version of `get_queryset` for a situation that never arises...

So maybe the right answer is to finally deliver on @kaedroho's promise from four years ago, and implement a configurable `include_root` flag :-)

Reply:
Ahh right, I forgot about the `.specific()` part. I can re-add this back in. The `exclude(depth=1)` seems to break when there's Page Chooser pagination that's why I originally removed it.